nixos/modules/services/web-apps/cryptpad.nix

```diff
@@ -0,0 +1,447 @@
+        extra bind mounts.
+      '';
+    };
+
+    config = mkOption {
+      description = mdDoc ''
+        cryptpad backend settings.
+        Some of the settings here (like backend ports) will be re-used for nginx config default as well
+        See https://github.com/cryptpad/cryptpad/blob/main/config/config.example.js for more extensive
+        comments and keys not described here
+        Note paths cannot be changed easily as they are also present indirectly in the nginx config
+        (for files that nginx serves directly)
+        Once setup, test your instance through `https://<domain>/checkup/`
+      '';
+      type = types.submodule {
+        freeformType = (pkgs.formats.json {}).type;
+        options = {
+          httpUnsafeOrigin = mkOption {
+            type = types.str;
+            example = "https://cryptpad.example.com";
+            description = mdDoc "Main cryptpad domain url";
+          };
+          httpSafeOrigin = mkOption {
+            type = types.nullOr types.str;
+            example = "https://cryptpad-sandbox.example.com. Apparently optional but recommended.";
+            description = mdDoc "Cryptpad sandbox url";
+          };
+          httpAddress = mkOption {
+            type = types.str;
+            default = "127.0.0.1";
+            description = mdDoc "Cryptpad listen address, also used in nginx to contact it";
+          };
+          httpPort = mkOption {
+            type = types.int;
+            default = 3000;
+            description = mdDoc "Backend main port";
+          };
+          httpSafePort = mkOption {
+            type = types.int;
+            default = 3001;
+            description = mdDoc "Backend sandbox port. Does not appear to be used...";
+          };
+          websocketPort = mkOption {
+            type = types.int;
+            default = 3003;
+            description = mdDoc "Backend websocket port";
+          };
+          maxWorkers = mkOption {
+            type = types.nullOr types.int;
+            default = null;
+            description = mdDoc "Number of child processes, defaults to number of cores available";
+          };
+          adminKeys = mkOption {
+            type = types.listOf types.str;
+            default = [];
+            description = mdDoc "Admins allowed to admin panel. Keys can be found in the settings page of the user";
+            example = [ "[cryptpad-user1@my.awesome.website/YZgXQxKR0Rcb6r6CmxHPdAGLVludrAF2lEnkbx1vVOo=]" ];
+          };
+          logToStdout = mkOption {
+            type = types.bool;
+            default = true;
+            description = mdDoc "Whether to log to stdout (will go to systemd service). Level can be changed with logLevel.";
+          };
+          blockDailyCheck = mkOption {
+            type = types.bool;
+            default = true;
+            description = mdDoc "Disable telemetry. This has priority over the 'Server telemetry' setting in admin panel.";
```

Comment:
I wasn't sure if telemetry should be disabled by default... Upstream enables it by default and doesn't even show the setting in the example config so it's a bit rude to turn on by default, as much as I'd love to.

Reply:
hmm, let's forget my complicated and probably incorrect English and try again:
 - Cryptpad's default has telemetry enabled (this isn't present in the config by default)
 - I'm torn between "respect upstream choice" and "I don't like telemetry to be opt-out"

The first version of this patch had this set to true (= telemetry disabled), but thinking it through I gave it and made it as upstream. Having the setting here in the config will hopefully make it easy enough to notice for anyone who cares.

On a side note, I just checked the code and I think I was wrong; the way the setting works in the admin menu, and that has priority alright. This setting is only effective until the admin menu has been clicked (that's a bit hard to fomulate... For example setting this to true (=disabled) will disable telemetry at start, but the menu will be showing telemetry enabled. Clicking to disable telemetry in the menu will check the box but not change anything; clicking again will re-enable telemetry)
I'll reformulate the comment to say this is the initial value until changed in the admin menu instead.

Reply:
Telemetry is always disabled by default in nixpkgs.